perceval/backends/core/jenkins.py

```diff
@@ -54,17 +54,28 @@ class Jenkins(Backend):
     :param sleep_time: time (in seconds) to sleep in case
         of connection problems
     :param archive: collect builds already retrieved from an archive
+    :param username: jenkins http username
+    :param password: jenkins http password
     """
     version = '0.11.2'
 
     CATEGORIES = [CATEGORY_BUILD]
 
-    def __init__(self, url, tag=None, archive=None,
+    def __init__(self, url, username=None, password=None, tag=None, archive=None,
                  blacklist_jobs=None, detail_depth=DETAIL_DEPTH, sleep_time=SLEEP_TIME):
         origin = url
 
         super().__init__(origin, tag=tag, archive=archive)
         self.url = url
+        if username is not None:
+            if password is not None:
+                if (url.startswith("http://")):
+                    self.url = url.replace('http://', 'http://' + username + ':' + password + '@')
+                elif (url.startswith("https://")):
+                    self.url = url.replace('https://', 'https://' + username + ':' + password + '@')
+            else:
+                logger.error("perceval: error:required argument --password is missing for user=%s", username)
+                exit(1)
```

Comment:
It's probably better to raise an exception instead of using `exit(1)`. Please,  have a look at: https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/git.py#L783, as example